core/src/html/text_format.rs

```diff
@@ -717,7 +717,34 @@ impl FormatSpans {
                             }
 
                             if let Some(size) = attribute(b"size") {
-                                format.size = size.parse().ok();
+                                // if the numer starts with + or -, the size is relative
+                                let (prefix, size) = if size.starts_with(&[b'+', b'-'][..]) {
+                                    (size.at(0) as u8, &size[1..])
+                                } else {
+                                    (0u8, &size[..])
+                                };
+
+                                // text is ignored from the first non-numeric character
+                                // (including the decimal separator)
+                                let first_foreign_char = size
+                                    .find(|c| c < '0' as u16 || c > '9' as u16)
+                                    .unwrap_or(size.len());
+                                let size = &size[0..first_foreign_char];
+
+                                let size: Option<f64> = size.parse().ok();
+
+                                if let Some(size) = size
+                                    .and_then(|size| match prefix {
+                                        b'+' => format.size.map(|last_size| last_size + size),
```

Comment:
Just to be sure, this means the size is relative to previous element, not to the parent?

Reply:
That means the size is relative to the parent. `format` is the current stack entry, a clone of the last stack entry (https://github.com/ruffle-rs/ruffle/pull/15077/files#diff-0313d265738726ac783fa404fb5ffff0bf42894a2a806e514185620b9a63251fR673), so I'm just applying the size difference to the would-be size (basically doing `format.size += size`/`format.size -= size` but with more checks).

In order to make the size relative to the previous element I would have to use `spans` here instead of the format stack.

Reply:
👍